### ISSUE TITLE
YOLOE with text prompts for detecting person + safety helmet: fine-tuning vs. custom data training? Few-image improvement with Roboflow?

### ISSUE THREAD

**Issue by dongdongju96** [ultralytics/ultralytics#22445]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Describe the question

I’d like to use **YOLOE text prompts** to detect `person` and `safety helmet` in my specific video environment. I want to adapt the model to my data and I’m unsure about the best approach:

* Is this considered **fine-tuning** or **custom data training** in YOLOE’s context?
* If I start from the pretrained weights (`yoloe-11s-seg.pt` or `yoloe-11l-seg.pt`) and add a **small number of labeled images** (via Roboflow) for additional training, should I expect measurable improvements?
* When I run something like:

  ```python
  target_classes = ["Person", "Safety Helmet"]
  text_embeddings = self.model.get_text_pe(target_classes)
  ```

  will text-prompting + these embeddings generally improve performance over the default class setup for this use case?

I’d also appreciate pointers to **docs, examples, or code paths** that explain how YOLOE uses text prompts and text embeddings internally (e.g., where prompt features are fused, how losses are computed when mixing text prompts with visual features, and recommended training configs).

### What I’m trying to achieve

* Reliable detection of:

  * `person`
  * `safety helmet`
* Robustness to my on-site conditions (camera angle, lighting, PPE colors, occlusion).
* Ideally, minimal additional labeling effort.

### What I’ve tried

* Inference with pretrained `yoloe-11s-seg.pt` and `yoloe-11l-seg.pt` using text prompts for the two classes above.

### Environment

Python 3.13.6
ultralytics==8.3.179
ultralytics-thop==2.0.15
torch==2.9.0+cu126
torchvision==0.24.0+cu126

### What I’m looking for

* Guidance on whether to prioritize **prompting** vs. **additional training** for this PPE scenario.
* Pointers to **docs/tutorials** and **internal code** related to YOLOE text prompts and training with custom data.

Thanks in advance for any direction!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @dongdongju96, thank you for your interest in Ultralytics 🚀! This is an automated response to help get things moving quickly — an Ultralytics engineer will also assist you here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

To help us guide you on prompting vs. additional training for your PPE scenario, please share:
- A brief description of your dataset (number of images, class distribution, example visuals)
- Exact commands or code snippets used for text prompts and inference/training
- Any validation metrics you’ve observed (precision/recall/mAP) with and without prompts
- The specific pretrained weights tested and your prompt list
- A few representative frames from your target environment (lighting, angles, PPE colors, occlusions)

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
If you have data, you can fine-tune YOLOE. You can't perform prompting after fine-tuning. It will only detect the classes you fine-tuned to detect.

If you have less data, use linear-probing. Otherwise, use normal fine-tuning.

https://docs.ultralytics.com/models/yoloe/#fine-tuning
